[SQL] unary minus (-) on INT UNSIGNED produces an error

Following SQL:

```
CREATE TABLE tbl(intt INTEGER UNSIGNED);

CREATE MATERIALIZED VIEW v AS SELECT
-intt AS intt
FROM tbl;
```

Given example input:
**"intt": 4294966290**

Produces the following error:
```
 error[E0600]: cannot apply unary operator `-` to type `u32`
  --> crates/feldera_pipe_operator_1e67241b63f64118/src/lib.rs:78:33
   |
78 |                 Some(x) => Some(-(x)),
   |                                 ^^^^ cannot apply unary operator `-`
   |
   = note: unsigned values cannot be negated
```

--- a/docs.feldera.com/docs/sql/integer.md
+++ b/docs.feldera.com/docs/sql/integer.md
@@ -6,7 +6,8 @@ bits).
 
 The legal operations are `+` (plus, unary and binary), `-` (minus,
 unary and binary), `*` (multiplication), `/` (division), `%`
-(modulus).
+(modulus).  Unary minus (negation) cannot be applied to unsigned
+values.
 
 Modulus involving negative numbers happens as follows:
 

--- a/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java
@@ -908,6 +908,11 @@ public class ExpressionCompiler extends RexVisitorImpl<DBSPExpression>
                 return makeUnaryExpression(node, type, DBSPOpcode.UNARY_PLUS, ops);
             case CHECKED_MINUS_PREFIX:
             case MINUS_PREFIX:
+                Utilities.enforce(ops.size() == 1);
+                DBSPType op0type = ops.get(0).getType();
+                if (op0type.is(DBSPTypeInteger.class) && !op0type.to(DBSPTypeInteger.class).signed) {
+                    throw new CompilationError("Unary minus cannot be applied to unsigned values", node);
+                }
                 return makeUnaryExpression(node, type, DBSPOpcode.NEG, ops);
             case BIT_AND:
                 return makeBinaryExpressions(node, type, DBSPOpcode.BW_AND, ops);

--- a/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/Regression1Tests.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/Regression1Tests.java
@@ -1717,4 +1717,15 @@ public class Regression1Tests extends SqlIoTest {
                 FROM v a
                 LEFT JOIN lookup l ON a.key = l.key;""");
     }
+    
+    @Test
+    public void issue5386() {
+        this.statementsFailingInCompilation("""
+                CREATE TABLE T(x INT UNSIGNED);
+                CREATE VIEW V AS SELECT -x FROM T;""",
+                "Unary minus cannot be applied");
+        this.statementsFailingInCompilation(
+                "CREATE VIEW V AS SELECT - CAST(1 AS INT UNSIGNED)",
+                "Unary minus cannot be applied");
+    }
 }

